github-actions/slash-commands/commands/rerun-circleci.ts

```diff
@@ -0,0 +1,110 @@
+import * as core from '@actions/core';
+import {Octokit} from '@octokit/rest';
+import {context} from '@actions/github';
+import {getAuthTokenForAngularRobotApp} from '../../utils';
+import fetch from 'node-fetch';
+import {RestEndpointMethodTypes} from '@octokit/plugin-rest-endpoint-methods';
+
+export async function rerunCircleCi() {
+  const token = await getAuthTokenForAngularRobotApp();
+  // Create authenticated Github client.
+  const github = new Octokit({auth: token});
+  /** The pull request for the action currently running. */
+  const {data: pullRequest} = await github.pulls.get({
+    ...context.repo,
+    pull_number: context.payload.issue!.number,
+  });
+
+  try {
+    const workflowId = await getCircleCiWorkflowIdForPullRequest(pullRequest, github);
+    const rerunUrl = `https://circleci.com/api/v2/workflow/${workflowId}/rerun?circle-token=${core.getInput(
+      'circleci-token',
+    )}`;
+    // Attempt to restart the rerun.
+    const response = await (
+      await fetch(rerunUrl, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({from_failed: true}),
+      })
+    ).json();
+
+    assertNoErrorsInCircleCiResponse(response);
+  } catch (err) {
+    // If the rerun attempt failed comment on the issue informing the requestor.
+    await github.issues.createComment({
+      ...context.repo,
+      issue_number: pullRequest.number,
+      body:
+        `@${context.actor} the CircleCI rerun you requested failed.  See details below:\n\n` +
+        `\`\`\`\n${err.message}\n\`\`\``,
```

Comment:
could we leak the CircleCI token by accident here? we might want to sanitize the output?

Reply:
I don't believe so, both as the auth headers are not included in outputs